optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStreamTest.java

```diff
@@ -788,6 +852,61 @@ public void nodeSharing() {
         assertEquals(1, monitorCount.get());
     }
 
-    // TODO add test for reusing streams: have 2 scorings from the same BiConstraintStream instance
+    @Test
+    public void filteredStreamReuse() {
```

Comment:
Can we rename these test method to reuseFilter() and reuseGroupBy()? it's first and foremost about reuse I'd argue